connectors/embeddings-vector-database/src/main/java/io/camunda/connector/embeddingstore/ClosableEmbeddingStore.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.embeddingstore;
+
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.EmbeddingSearchRequest;
+import dev.langchain4j.store.embedding.EmbeddingSearchResult;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.filter.Filter;
+import java.util.Collection;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A wrapper for {@link EmbeddingStore} that implements {@link AutoCloseable} to ensure proper
+ * resource management. This class delegates all embedding store operations to the wrapped instance.
+ *
+ * <p>Usage example:
+ *
+ * <pre>{@code
+ * try (ClosableEmbeddingStore store = ClosableEmbeddingStore.from(embeddingStore)) {
+ *     store.add(embedding, textSegment);
+ *     // store will be automatically closed
+ * }
+ * }</pre>
+ */
+public class ClosableEmbeddingStore implements EmbeddingStore<TextSegment>, AutoCloseable {
```

Comment:
💭 Just a thought, might not really make a difference: could this also be a generic `ClosableEmbeddingStore<T> implements EmbeddingStore<T>, AutoCloseable`, so it would work for other use cases than `TextSegment`?

Reply:
I intentionally made it specific to `TextSegment` for simplicity as we currently only work with `TextSegment`. We can make it generic when we support multi-modal. But I am not against making it generic right now either.